tensorflow_federated/python/core/impl/executors/federating_executor.py

```diff
@@ -453,6 +451,221 @@ def validate_executor_placements(cls, executor_placements):
+            arg.type_signature))
+    py_typecheck.check_type(arg.internal_representation,
+                            anonymous_tuple.AnonymousTuple)
+    py_typecheck.check_len(arg.internal_representation, 5)
+
+    # Note: This is a simple initial implementation that simply forwards this
+    # to `federated_reduce()`. The more complete implementation would be able
+    # to take advantage of the parallelism afforded by `merge` to reduce the
+    # cost from liner (with respect to the number of clients) to sub-linear.
+
+    # TODO(b/134543154): Expand this implementation to take advantage of the
+    # parallelism afforded by `merge`.
+    fed_ex = self.federating_executor
+
+    val = arg.internal_representation[0]
+    zero = arg.internal_representation[1]
+    accumulate = arg.internal_representation[2]
+    pre_report = await fed_ex._compute_intrinsic_federated_reduce(
+        FederatingExecutorValue(
+            anonymous_tuple.AnonymousTuple([(None, val), (None, zero),
+                                            (None, accumulate)]),
+            computation_types.NamedTupleType(
+                (val_type, zero_type, accumulate_type))))
+
+    py_typecheck.check_type(pre_report.type_signature,
+                            computation_types.FederatedType)
+    type_utils.check_equivalent_types(pre_report.type_signature.member,
+                                      report_type.parameter)
+
+    report = arg.internal_representation[4]
+    return await fed_ex._compute_intrinsic_federated_apply(
+        FederatingExecutorValue(
+            anonymous_tuple.AnonymousTuple([
+                (None, report), (None, pre_report.internal_representation)
+            ]),
+            computation_types.NamedTupleType(
+                (report_type, pre_report.type_signature))))
+
+  async def federated_sum(self, arg):
+    py_typecheck.check_type(arg.type_signature, computation_types.FederatedType)
+    zero, plus = tuple(await asyncio.gather(*[
+        executor_utils.embed_tf_scalar_constant(
+            self.federating_executor,
+            arg.type_signature.member,
+            0),
+        executor_utils.embed_tf_binary_operator(
+            self.federating_executor,
+            arg.type_signature.member,
+            tf.add)]))
+    return await self.federating_executor._compute_intrinsic_federated_reduce(
+        FederatingExecutorValue(
+            anonymous_tuple.AnonymousTuple([
+                (None, arg.internal_representation),
+                (None, zero.internal_representation),
+                (None, plus.internal_representation)
+            ]),
+            computation_types.NamedTupleType(
+                (arg.type_signature, zero.type_signature, plus.type_signature)))
+    )
+
+  async def federated_mean(self, arg):
+    # TODO check this logic, just switched SERVER to AGGREGATOR
+    arg_sum = await self.federating_executor._compute_intrinsic_federated_sum(arg)
```

Comment:
afaict the placement for the sum is the server, not the aggregator, so when you scale below the placements won't match up -- maybe it's just a case of using wrong placement for the result of `federated_reduce`

Reply:
Why do you say that?

Reply:
When I asked about where the result of reduce should be, we said it would be the server and not the aggregator. Changing that would be a fix, right?

Reply:
per federated intrinsic the result should probably end up on the server; issue here is that the same doesn't/shouldn't hold for intermediate values

Reply:
but this means we will have to reimplement the whole reduce op just for the mean -- that doesn't seem like the right solution either. any suggestions?

Reply:
could you pass a placement to the `_compute_intrincis` functions, specifying where you want the result to end up?